_posts/2024-02-14-provisioning-ubuntu-autoinstall.md

```diff
@@ -0,0 +1,361 @@
+This means that you have to use `Preseed` templates suitable for the `debian-installer` mechanism.
+- For Ubuntu with major OS version `20.04` with minor OS version `3` or above, or `22.04.z`: Foreman searches for `initrd` and `vmlinuz` in `casper/`.
+This means that you have to use `Autoinstall` templates suitable for the `Subiquity` installer.
+
+When you create a host, Foreman retrieves the boot files from the URL of the installation media.
+For Debian and Ubuntu prior to 20.04.3, you could use `http://archive.ubuntu.com` and Foreman would find the required files based on the code name configured for the OS entry.
+For Ubuntu 20.04.3+, those files are no longer present in the official upstream repositories.
+Therefore, you have to extract the ISO image and make it accessible for Foreman Server or Smart Proxy Server during provisioning.
+For example, you can place it in `FQDN/pub/installation_media/ubuntu/20.04/`.
+
+### Provisioning Debian/Ubuntu Compared to Enterprise Linux
+
+In contrast to Enterprise Linux such as Red Hat Enterprise Linux or AlmaLinux, you cannot provision Ubuntu based on synchronized content.
+This is due to the missing Linux Kernel and initrd files in the synchronized Pulp repositories.
+The alternative for both Debian and Ubuntu has mostly been to either use the upstream repositories (downside: you need access to the internet), or by extracting the installation files from the ISO image and providing them under `pub/` on your Foreman Server (downside: depends on the z-release).
+
+### Provisioning process
+
+- Foreman provides the rendered user data template on `FQDN/userdata/`.
+This interface is already in use for all image-based deployments that use `cloud-init` templates.
+- Because the ISO image is also required for the deployment but Foreman does not provide any dedicated interface for this, we have documented a way to place the ISO image into the `FQDN/pub/installation_media` directory on Foreman Server.
+This means Foreman serves the ISO image over `http`.
+If you want to deploy a host from a Smart Proxy Server, you have to copy the ISO image to your Smart Proxy Server.
+- Ubuntu Autoinstall deployments also require the extracted ISO image.
+You have to place the extracted ISO image in `FQDN/pub/installation_media/` on your Foreman Server and name the directory identical to the ISO image without the `.iso` file ending.
+If you want to deploy a host from a Smart Proxy Server, you have to copy the extracted ISO image to your Smart Proxy Server.
+
+Ensure that you provide the ISO and extracted ISO under the same name, for example `22_04.iso` and `22_04/`, to use the URL for the installation media as path for both.
+For more information, see [Creating an Installation Medium for Ubuntu 22.04](https://docs.theforeman.org/3.9/Provisioning_Hosts/index-katello.html#Creating_an_Installation_Medium_for_Ubuntu_22_04_provisioning) in _Provisioning Hosts_.
+
+#### preseed_kernel_options_autoinstall.erb
```

Comment:
I fail to see why this code snippet is here and what the reader should learn from it.

Reply:
To illustrate the new kernel options and the way the ISO is "found/expected" based on the path of the installation media. I would also be OK dropping it (to shorten the blog article).

Reply:
Ah. Maybe intro that with a line or two of text? Something like "The enablement of the autoinstall process and the configuration of the path to the ISO etc happens via kernel options that are set in the `blah` snippet:"?